x-pack/legacy/plugins/fleet/server/libs/agent.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+import { Agent, SortOptions, NewAgent } from './adapters/agent/adapter_type';
+import { AgentAdapter } from './adapters/agent/default';
+import { TokenLib } from './token';
+
+export class AgentLib {
+  constructor(private readonly agentAdater: AgentAdapter, private readonly tokens: TokenLib) {}
+
+  /**
+   * Enroll a new token into elastic fleet
+   */
+  public async enroll(token: any, agentData: NewAgent): Promise<Agent> {
```

Comment:
Open question what happen if we enroll two time the same agent: (same `shared_id`)

Reply:
Great question, we return an error if it is active already. If it is not active, we re-activate it returning a new authentication token